### PR TITLE
Example with MySQL and Bootstrap 4 for responsive web

### PR DESCRIPTION
**Pull request by mmaguero**
Hello,

This is an example with a MySQL database, using Bootstrap 4 for the sensitive web.

Modify it as you want to add it to the sample folder.

Thank you very much, the complement is fabulous!

### CODE REVIEW

File: Makefile.PL [hunk 1]
```diff
@@ -20,6 +20,8 @@ WriteMakefile(
         'HTML::Table::FromDatabase' => 1.10,
         'CGI::FormBuilder' => 3.10,
         'DBD::SQLite'   => 1.50,
+        'DBD::mysql' => 0,
```
Comment (joshrabinowitz) [inline Makefile.PL]: not sure we should make DBD::mysql a hard requirement if it's only used in an example

File: example/mysql/README.pod [hunk 2]
```diff
@@ -0,0 +1,31 @@
+=head1 NAME
+
+Dancer::Plugin::SimpleCRUD example application with MySQL and Bootstrap 4
+
+=head1 DESCRIPTION
+
+A simple L<Dancer> application to demonstrate L<Dancer::Plugin::SimpleCRUD> in action with MySQL and Bootstrap 4 for responsive webs
```
Comment (joshrabinowitz) [inline example/mysql/README.pod]: typo at end of line?

File: example/mysql/README.pod [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+=head1 USAGE
+
```
Comment (joshrabinowitz) [inline example/mysql/README.pod]: Perhaps mention using schema.sql to create the db and table?

File: example/mysql/schema.sql [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+INSERT INTO people VALUES(8,'Sophie','Doe','sophie.doe@example.com','Female',21,'',1,0);
+INSERT INTO people VALUES(9,'Hayley','Doe','hayley.doe@example.com','Female',19,NULL,1,0);
+INSERT INTO people VALUES(10,'Michelle','Doe','michelle.doe@example.com','Female',29,NULL,1,0);
+INSERT INTO people VALUES(11,'Bob','Doe','bob.doe@example.com','Male',36,'Surprise - another test!',1,0);
+INSERT INTO people VALUES(12,'Burt','Bacharach','burt@example.com','Male',84,'',2,0);
+CREATE TABLE employer ( id integer NOT NULL AUTO_INCREMENT primary key, name varchar(150) );
+INSERT INTO employer VALUES(1,'ACME Ltd');
+INSERT INTO employer VALUES(2,'Bob''s Widgets Ltd');
+INSERT INTO employer VALUES(3,'Bertie''s Badgers PLC');
+COMMIT;
```
Comment (joshrabinowitz) [inline example/mysql/schema.sql]: I don't think this COMMIT belongs here as we're not in a transaction?

File: example/sqlite/views/contentwrapper.tt [hunk 5]
```diff
@@ -0,0 +1 @@
+<% render_content %>
```
Comment (joshrabinowitz) [inline example/sqlite/views/contentwrapper.tt]: Why is this file added here?

File: Makefile.PL (hunk 1 above)
Comment (mmaguero) [inline Makefile.PL]: Agreed!

File: example/mysql/README.pod (hunk 2 above)
Comment (mmaguero) [inline example/mysql/README.pod]: I have corrected it

File: example/mysql/schema.sql (hunk 4 above)
Comment (mmaguero) [inline example/mysql/schema.sql]: Agreed!

File: example/sqlite/views/contentwrapper.tt (hunk 5 above)
Comment (mmaguero) [inline example/sqlite/views/contentwrapper.tt]: I have corrected it

File: example/mysql/views/layouts/main.tt [hunk 6]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+<title>Management of company users</title>
```
Comment (joshrabinowitz) [inline example/mysql/views/layouts/main.tt]: Can we parameterize this so it shows the correct table/record names if used with a different simple_crud() call?

File: example/mysql/views/layouts/main.tt [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+<!-- Nav Content -->
+<div class="collapse navbar-toggleable-xs" id="nav-content">
+<a class="navbar-brand" href="#">Manage company users</a>
+<ul class="nav navbar-nav">
+<li class="nav-item">
+<a class="nav-link" href="/people">Home</a>
+</li>
+<li class="nav-item">
+<a class="nav-link" href="/people/add">Add</a>
```
Comment (joshrabinowitz) [inline example/mysql/views/layouts/main.tt]: can we parameterize  "Manage company user" titles and the /people links so they works correctly for other simple_crud() uses? Ideally this template could be used for other simple_crud calls out of the box.

File: example/mysql/views/layouts/main.tt [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+</nav>
+
+<div class="container">
+
+<div class="bs-example">
+<% content %>
+</div>
+
+<br />
+<pre><small>Provided by Simple::CRUD::Plugin (PerlDancer 1.3 + Bootstrap 4)</small></pre>
```
Comment (joshrabinowitz) [inline example/mysql/views/layouts/main.tt]: Small issue: seems like these version numbers may get out of date over time.

Comment (joshrabinowitz) [inline example/mysql/views/layouts/main.tt]: " Dancer::Plugin::SimpleCRUD", not Simple::CRUD::Plugin

Comment (joshrabinowitz) [inline example/mysql/views/layouts/main.tt]: "Dancer::Plugin::SimpleCRUD" not Simple::CRUD::Plugin

**Comment by bigpresh** [comment]
I've been super-busy, but just wanted to comment to say big thanks @mmaguero for the awesome contribution and @joshrabinowitz for reviewing it!

File: example/mysql/simplecrud-example.pl [hunk 9]
```diff
@@ -11,6 +11,7 @@
     prefix => '/people',
     acceptable_values => {
         gender => [ qw( Male Female ) ],
+		is_admin => [ qw( 0 1 ) ],
```
Comment (joshrabinowitz) [inline example/mysql/simplecrud-example.pl]: can we make this line up with 'gender'?

File: example/mysql/views/layouts/main.tt [hunk 10]
```diff
@@ -17,13 +17,13 @@
...
-<a class="navbar-brand" href="#">Manage company users</a>
+<a class="navbar-brand" href="#"><% settings.appname %></a>
 <ul class="nav navbar-nav">
 <li class="nav-item">
-<a class="nav-link" href="/people">Home</a>
+<a class="nav-link" href="<% request.uri_base %>">Home</a>
 </li>
 <li class="nav-item">
-<a class="nav-link" href="/people/add">Add</a>
+<a class="nav-link" href="/add">Add</a>
```
Comment (joshrabinowitz) [inline example/mysql/views/layouts/main.tt]: Seems like using /people/add is the overall simplest thing here. Ideally this would work totally dynamically for other table types, but that's not needed for just this sample code.

I think the "get /add" route just makes things more difficult overall for the future when the user wants to add more simplecrud interfaces. .

**Comment by joshrabinowitz** [comment]
This really looks nice! Thanks for adding the example mmageuro!

File: example/mysql/simplecrud-example.pl [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+);
+
+get '/' => sub {
+    redirect '/people';
+};
+
+# redirect for add new register in db_table arg
+get '/add' => sub {
+    redirect '/people/add';
+};
```
Comment (joshrabinowitz) [inline example/mysql/simplecrud-example.pl]: I think these redirects make it difficult for people to use the provided template and add more simplecrud interfaces.

**Comment by bigpresh** [comment]
Just wanted to say that I haven't forgotten this PR!

I was hoping to condense the SQLite and MySQL ones into one example, as at the end of the day, D::P::SC is supposed to be DBMS-agnostic as far as possible, and the SQLite example is easier to test out with a single command; I imagine people who're trying out an example will mostly choose whatever is quickest and easiest to spin up, and a script that uses DBD::SQLite with a provided schema.sql to Just Work is pretty quick and easy :)  In particular, I'm not sure that duplicating the templates etc between the two is entirely warranted.

As soon as I get a chance I plan to test this out fully :)
